ical.NET/Utility/SerializationUtil.cs

```diff
@@ -49,11 +50,11 @@ private static List<MethodInfo> GetDeserializingMethods(Type targetType)
                 .Select(t => t.targetTypeMethodInfo)
                 .ToList();
 
-            _onDeserializingMethods.Add(targetType, methodInfo);
+            _onDeserializingMethods.AddOrUpdate(targetType, methodInfo, (type, list) => methodInfo);
```

Comment:
Since you're doing an unconditional update, you could just do `_onDeserializingMethods[targetType] = methodInfo;`.  

I think what you really want here is a single call to `GetOrAdd`, passing a lambda that does the calculation.  That is: 

```csharp
return _onDeserializingMethods.AddOrUpdate.GetOrAdd(targetType, tt => tt
    .GetMethods(_bindingFlags)
    .Where(targetTypeMethodInfo => targetTypeMethodInfo
        .GetCustomAttributes(typeof(OnDeserializingAttribute), false).Any())
    .ToList());
```

Using the lambda parameter `tt` instead of the method parameter `targetType` means the lambda won't capture an environment, so the compiler can cache it in a static field.